test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -420,30 +420,30 @@ func setupCRDAndVerifySchema(f *framework.Framework, schema, expect []byte, grou
 		return nil, fmt.Errorf("require at least one version for CRD")
 	}
 
-	props := &v1beta1.JSONSchemaProps{}
+	props := &apiextensionsv1.JSONSchemaProps{}
 	if schema != nil {
 		if err := yaml.Unmarshal(schema, props); err != nil {
 			return nil, err
 		}
 	}
 
-	crd, err := crd.CreateMultiVersionTestCRD(f, group, func(crd *v1beta1.CustomResourceDefinition) {
-		var apiVersions []v1beta1.CustomResourceDefinitionVersion
+	crd, err := crd.CreateMultiVersionTestCRD(f, group, func(crd *apiextensionsv1.CustomResourceDefinition) {
+		var apiVersions []apiextensionsv1.CustomResourceDefinitionVersion
 		for i, version := range versions {
-			apiVersions = append(apiVersions, v1beta1.CustomResourceDefinitionVersion{
+			version := apiextensionsv1.CustomResourceDefinitionVersion{
 				Name:    version,
 				Served:  true,
 				Storage: i == 0,
-			})
-		}
-		crd.Spec.Versions = apiVersions
-
-		// set up validation when input schema isn't nil
-		if schema != nil {
-			crd.Spec.Validation = &v1beta1.CustomResourceValidation{
-				OpenAPIV3Schema: props,
 			}
+			// set up validation when input schema isn't nil
+			if schema != nil {
```

Comment:
schema's required in v1, so if we weren't given one, we have to set it to something. `{Type: "object", XPreserveUnknownFields: true}` is the minimal data-preserving schema we accept

Reply:
Thanks, I just ran into this when testing. Fixed all occurrences.